repository.go

```diff
@@ -389,6 +391,66 @@ func (v *Repository) CreateCommit(
 	return oid, nil
 }
 
+func (v *Repository) CreateCommitFromIds(
+	refname string, author, committer *Signature,
+	message string, tree *Oid, parents ...*Oid) (*Oid, error) {
+
+	oid := new(Oid)
+
+	var cref *C.char
+	if refname == "" {
+		cref = nil
+	} else {
+		cref = C.CString(refname)
+		defer C.free(unsafe.Pointer(cref))
+	}
+
+	cmsg := C.CString(message)
+	defer C.free(unsafe.Pointer(cmsg))
+
+	var parentsarg **C.git_oid = nil
+
+	nparents := len(parents)
+	if nparents > 0 {
+		parentsarg = (**C.git_oid)(C.malloc(C.size_t(unsafe.Sizeof(uintptr(0)) * uintptr(nparents))))
+		defer C.free(unsafe.Pointer(parentsarg))
+		parentsptr := uintptr(unsafe.Pointer(parentsarg))
+		for _, v := range parents {
+			*(**C.git_oid)(unsafe.Pointer(parentsptr)) = v.toC()
+			parentsptr += unsafe.Sizeof(uintptr(0))
```

Comment:
Can we avoid this pointer arithmetic? Can we generate the array in Go and pass that pointer to `C.git_commit_create_from_ids`?

Reply:
IIRC this was a workaround for the dreaded [cgo argument has Go pointer to Go pointer](https://stackoverflow.com/questions/51483032/passing-go-pointer-to-cgo). admittedly, i didn't invest too much time trying more options, so let me give it another try.

Reply:
Tried different things, but now I'm pretty sure we cannot avoid this :( added a comment.